packages/core/src/command/consolidated/consolidated-filter-by-resource.ts

```diff
@@ -0,0 +1,25 @@
+import { Bundle } from "@medplum/fhirtypes";
+import { ResourceTypeForConsolidation } from "@metriport/shared/medical";
+
+/**
+ * Filters resources in a FHIR Bundle by date.
+ *
+ * @param bundle The FHIR Bundle to filter.
+ * @param dateFrom The start date for filtering (inclusive).
+ * @param dateTo The end date for filtering (inclusive).
+ * @returns A new FHIR Bundle based on the original one, containing only the filtered resources.
+ */
+export function filterBundleByResource(
+  bundle: Bundle,
+  resources?: ResourceTypeForConsolidation[] | undefined
+): Bundle {
+  if (!resources || resources.length === 0) {
+    return bundle;
+  }
+  const filtered =
+    bundle.entry?.filter(entry => {
```

Comment:
Why is entry undefined? If it can be, why return the empty list if it is (and why not just leave it undefined?).

Reply:
It can be undefined b/c that's the type definition of FHIR (and our types). I'm setting to empty array instead of undefined out of practice (avoid undefined when possible), but can change if you feel strong about it. :)

Reply:
No worries 👍